File: examples/renewal-rs/src/parameters.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
pub struct Parameters {
    pub r0: f64,
    pub generation_interval_pmf: Vec<f64>,
    pub symptom_onset_pmf: Vec<f64>,
    pub initial_infections: Vec<u64>,
    pub sim_length: usize,
    pub population: Option<u64>,
    pub seed: u64,
}

File: examples/renewal/src/renewal/parameters.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Parameters:
    r0: float
    generation_interval_pmf: list[float]
    symptom_onset_pmf: list[float]
    initial_infections: list[int]
    sim_length: int
    population_size: int | None = None
